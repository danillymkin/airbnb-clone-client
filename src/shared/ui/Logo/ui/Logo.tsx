'use client';

import React, { FunctionComponent } from 'react';
import Image from 'next/image';

interface LogoProps {}

type Props = LogoProps;

export const Logo: FunctionComponent<Props> = ({}): JSX.Element => {
  return (
    <Image
      alt={'Logo'}
      className={'hidden cursor-pointer md:block'}
      height={31}
      width={100}
      src={'/images/logo.png'}
      priority
    />
  );
};
