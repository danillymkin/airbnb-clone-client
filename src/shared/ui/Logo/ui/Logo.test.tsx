import { Logo } from './Logo';
import { render, screen } from '@testing-library/react';

describe('Logo', () => {
  it('should render logo', () => {
    render(<Logo />);
    const logo = screen.getByRole('img');
    expect(logo).toBeTruthy();
  });

  it('should match snapshot', () => {
    render(<Logo />);
    const logo = screen.getByRole('img');
    expect(logo).toMatchSnapshot();
  });

  it('should have right alt', () => {
    render(<Logo />);
    const logo = screen.getByAltText(/logo/i);
    expect(logo).toBeTruthy();
  });

  it('should have right src', () => {
    render(<Logo />);
    const logo = screen.getByRole('img');
    expect(logo.getAttribute('src')).toContain('logo.png');
  });
});
