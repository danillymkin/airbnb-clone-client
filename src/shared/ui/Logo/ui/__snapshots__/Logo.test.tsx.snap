// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match snapshot 1`] = `
<img
  alt="Logo"
  class="block cursor-pointer"
  data-nimg="1"
  decoding="async"
  fetchpriority="high"
  height="31"
  src="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=75"
  srcset="/_next/image?url=%2Fimages%2Flogo.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=256&q=75 2x"
  style="color: transparent;"
  width="100"
/>
`;
